packages/grafana-prometheus/src/components/metrics-browser/useMetricsLabelsValues.ts

```diff
@@ -0,0 +1,386 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+import { TimeRange } from '@grafana/data';
+
+import PromQlLanguageProvider from '../../language_provider';
+
+import { buildSelector } from './selectorBuilder';
+import { DEFAULT_SERIES_LIMIT, EMPTY_SELECTOR, LAST_USED_LABELS_KEY, Metric, METRIC_LABEL } from './types';
+
+export const useMetricsLabelsValues = (timeRange: TimeRange, languageProvider: PromQlLanguageProvider) => {
+  const timeRangeRef = useRef<TimeRange>(timeRange);
+  // const [initTrigger, setInitTrigger] = useState(Date.now());
+
+  const [seriesLimit, setSeriesLimit] = useState(DEFAULT_SERIES_LIMIT);
+  const [err, setErr] = useState('');
+  const [status, setStatus] = useState('Ready');
+  const [validationStatus, setValidationStatus] = useState('');
+
+  const [metrics, setMetrics] = useState<Metric[]>([]);
+  const [selectedMetric, setSelectedMetric] = useState('');
+  const [labelKeys, setLabelKeys] = useState<string[]>([]);
+  const [selectedLabelKeys, setSelectedLabelKeys] = useState<string[]>([]);
+  const [lastSelectedLabelKey, setLastSelectedLabelKey] = useState('');
+  const [labelValues, setLabelValues] = useState<Record<string, string[]>>({});
+  const [selectedLabelValues, setSelectedLabelValues] = useState<Record<string, string[]>>({});
+
+  // We don't want to trigger fetching for small amount of time changes.
+  // When MetricsBrowser re-renders for any reason we might receive a new timerange.
+  // This particularly happens when we have relative time ranges: from: now, to: now-1h
+  useEffect(() => {
+    if (
+      timeRange.to.diff(timeRangeRef.current.to, 'second') >= 5 &&
+      timeRange.from.diff(timeRangeRef.current.from, 'second') >= 5
+    ) {
+      timeRangeRef.current = timeRange;
+    }
+  }, [timeRange]);
+
+  //Handler for error processing - logs the error and updates UI state
+  const handleError = useCallback((e: unknown, msg: string) => {
+    console.error(e);
```

Comment:
I would remove this condole error as we are showing it in the UI. Maybe only console.error if it us unknown error. 🤔